lib/serialDevice.js

```diff
@@ -266,11 +266,14 @@ module.exports = function(config) {
 
           // TODO: Dexcom-specific code to be removed,
           // once OSX El Capitan is updated to not require Dexcom restart
-          if((this.app.state._os === 'mac') && (deviceInfo.driverId === 'Dexcom') && (deviceInfo.path.match(deviceInfo.portPattern) === null)) {
-            return cb(new Error('You may also need to power off and on your Dexcom receiver. ' + deviceDebugInfo));
-          }
+          chrome.runtime.getPlatformInfo(function(platformInfo) {
+            if ((platformInfo.os === 'mac') && (deviceInfo.driverId === 'Dexcom') &&
+              (deviceInfo.path.match(deviceInfo.portPattern) === null)) {
+              return cb(new Error('You may also need to power off and on your Dexcom receiver. ' + deviceDebugInfo));
+            }
 
-          cb(new Error('Could not connect to a matching device port: ' + deviceDebugInfo));
+            cb(new Error('Could not connect to a matching device port: ' + deviceDebugInfo));
+          });
```

Comment:
Yeah, this is the thing I thought you might find :)

Re: the location of the callback, I moved it in because otherwise it could get called too early, no? Since the `getPlatformInfo` is itself apparently asynchronous? It seems to still work in the desired circumstances - I get the 'Could not connect to matching device port' error details when clicking upload on Tandem with no Tandem plugged in, for example.

And indeed, we can't read the os state here the way we used to because before the app state was bound to a global app object, and honestly that was a hack (essentially to get around the clean separation of concerns). It's not a best practice, and I didn't want to continue it with redux, especially not here since this additional check is temporary, right? Once Chrome 48 is widely adopted we'll have better ways to distinguish the Dexcom error types?